test/export/test_passes.py

```diff
@@ -213,5 +215,100 @@ def forward(self, x, y):
         self.assertEqual(gm_result_for_1_size, eager_result_for_1_size)
 
 
+    def test_runtime_assert_inline_constraints_for_item(self) -> None:
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                b = x.item()
+                constrain_as_value(b, min=2, max=5)
+                return b
+
+        x = torch.tensor([2])
+        mod = M()
+        gm = _export(mod, (x,))
+
+        pass_result = AddRuntimeAssertionsForConstraintsPass()(gm)
+        new_gm = pass_result.graph_module
+        self.assertTrue(pass_result.modified)
+
+        num_assert = count_call_function(new_gm.graph, torch.ops.aten._assert_async.msg)
+        num_scalar_tensor = count_call_function(new_gm.graph, torch.ops.aten.scalar_tensor.default)
+        # 1 constraint for shape of x, 2 constraints for b
+        self.assertEqual(num_assert, 3)
+        self.assertEqual(num_scalar_tensor, 3)
+
+        with self.assertRaisesRegex(RuntimeError, r"^_local_scalar_dense_default.*\[2, 5\].$"):
```

Comment:
This error message is bit hard to understand from user's perspective, how about we just dump the stacktrace from user code instead?